R/CallbackSetLRScheduler.R

```diff
@@ -41,15 +41,31 @@ CallbackSetLRScheduler = R6Class("CallbackSetLRScheduler",
 
       self$scheduler_fn = .scheduler
       private$.scheduler_args = list(...)
+
       if (step_on_epoch) {
-        self$on_epoch_end = function() self$scheduler$step()
+        if (class(self$scheduler_fn)[[1L]] == "lr_reduce_on_plateau") {
```

Comment:
I would suggest we solve this differently.
I would probably just implement a custom sub-class for the `lr_reduce_on_plateu` scheduler.